contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6508,6 +6509,8 @@ pltsql_remove_current_query_env(void)
 		AbortCurTransaction = false;
 
 		ENRDropTempTables(currentQueryEnv);
+		if (is_abort)
+			smgrDoPendingDeletes(!old_abort_curr_txn);
```

Comment:
No, this should not have any effect. Our TV handling just makes sure that TVs always get dropped no matter what, whether we abort or commit the transaction - this change won't effect that behavior.

Reply:
Hmm. `smgrDoPendingDeletes()` would handle non-temp tables also, correct? Did we test the case where `smgrDoPendingDeletes` could potentially drop files for non-temporary tables? My main concern is that the tables we're dropping in `ENRDropTempTables` may not match exactly with what `smgrDoPendingDeletes` is deleting.

Reply:
Yes, it is technically possible - however, this code path is only hit at the end of either a stored procedure call or a batch. Therefore, it will follow the normal transaction semantics under most circumstances, and only need to do this special cleanup at transaction boundaries. I believe that this is backed up by the fact the fact that our full JDBC test suite succeeded. I'll try some ad-hoc testing to exercise this further, but I am confident in this solution.

Reply:
I did some testing mixing temp and permanent tables, and they all behaved normally:
```
1> create proc proc1 as begin; declare @tv table (a int); create table #t1 (a int); insert into t1 values (1); insert into @tv values (1); insert into #t1 values (1); end;
2> go
1> begin tran
2> insert into t1 values (2)
3> exec proc1
4> select * from t1
5> drop table t1
6> rollback
7> go

(1 rows affected)

(1 rows affected)

(1 rows affected)

(1 rows affected)
a          
-----------
          2
          1

(2 rows affected)
1> select * from t1
2> go
a          
-----------

(0 rows affected)
1> begin tran
2> insert into t1 values (2)
3> create table #outer_table (a int)
4> insert into #outer_table values (1)
5> exec proc1
6> select * from t1
7> drop table t1
8> rollback
9> go

(1 rows affected)

(1 rows affected)

(1 rows affected)

(1 rows affected)

(1 rows affected)
a          
-----------
          2
          1

(2 rows affected)
1> select * from t1
2> go
a          
-----------

(0 rows affected)
1> select * from #outer_table
2> go
Msg 33557097, Level 16, State 1, Server BABELFISH, Line 1
relation "#outer_table" does not exist
```